plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/function/tablechanges/DeltaLakeTableChangesFunction.java

```diff
@@ -0,0 +1,190 @@
+    public static final String SCHEMA_NAME = "system";
+    public static final String NAME = "table_changes";
+    public static final SchemaFunctionName TABLE_CHANGES_NAME = new SchemaFunctionName(SCHEMA_NAME, NAME);
+    public static final String CHANGE_TYPE_COLUMN_NAME = "_change_type";
+    public static final String COMMIT_VERSION_COLUMN_NAME = "_commit_version";
+    public static final String COMMIT_TIMESTAMP_COLUMN_NAME = "_commit_timestamp";
+
+    private final HiveMetastoreFactory hiveMetastoreFactory;
+    private final TransactionLogAccess transactionLogAccess;
+    private final TypeManager typeManager;
+    private final CachingExtendedStatisticsAccess statisticsAccess;
+    private final TrinoFileSystemFactory fileSystemFactory;
+
+    public DeltaLakeTableChangesFunction(
+            HiveMetastoreFactory hiveMetastoreFactory,
+            TransactionLogAccess transactionLogAccess,
+            TypeManager typeManager,
+            CachingExtendedStatisticsAccess statisticsAccess,
+            TrinoFileSystemFactory fileSystemFactory)
+    {
+        super(
+                SCHEMA_NAME,
+                NAME,
+                ImmutableList.of(
+                        ScalarArgumentSpecification.builder().name(TABLE_NAME_ARGUMENT).type(VARCHAR).build(),
+                        ScalarArgumentSpecification.builder().name(START_VERSION_ARGUMENT).type(BIGINT).defaultValue(-1L).build(),
+                        ScalarArgumentSpecification.builder().name(END_VERSION_ARGUMENT).type(BIGINT).defaultValue(Long.MAX_VALUE).build()),
+                GENERIC_TABLE);
+        this.hiveMetastoreFactory = requireNonNull(hiveMetastoreFactory, "hiveMetastoreFactory is null");
+        this.transactionLogAccess = requireNonNull(transactionLogAccess, "transactionLogAccess is null");
+        this.typeManager = requireNonNull(typeManager, "typeManager is null");
+        this.statisticsAccess = requireNonNull(statisticsAccess, "statisticsAccess is null");
+        this.fileSystemFactory = requireNonNull(fileSystemFactory, "fileSystemFactory is null");
+    }
+
+    @Override
+    public TableFunctionAnalysis analyze(ConnectorSession session, CatalogHandle catalogHandle, ConnectorTransactionHandle transaction, Map<String, Argument> arguments)
+    {
+        ScalarArgument tableNameArgument = (ScalarArgument) arguments.get(TABLE_NAME_ARGUMENT);
+        requireNonNull(tableNameArgument.getValue(), "tableName value for function table_changes() is null");
+        String schemaTableName = ((Slice) tableNameArgument.getValue()).toStringUtf8();
+        if (!schemaTableName.contains(".") || schemaTableName.indexOf(".") != schemaTableName.lastIndexOf(".")) {
+            throw new IllegalArgumentException("table_name value for function table_changes() must be in following format: schema_name.table_name");
+        }
+
+        ScalarArgument startVersionArgument = (ScalarArgument) arguments.get(START_VERSION_ARGUMENT);
```

Comment:
If the start version is not available, I'm assuming that the first version to be read is the first version available in the history.

See `delta.logRetentionDuration` setting.

Reply:
I think the default we want is the beginning of the table, not the first version available. If it's been expired by the log retention then throw an exception. Does that sound right?

Reply:
hmm i didn't think about that, i will see what databricks does in such a situation

Reply:
So deltabricks doesn't allow this, they always expect  a startVersion parameter and if it is not there the query fails, if you provide 0 but you did run vacuum then it will also fail. 
What should we do here? Start from first available version or try starting from 0 ? @findinpath @alexjo2144

Reply:
I am pretty sure that we want to start from version 0 when you don't provide a start version and fail if the files to do that are not available.

Reply:
so this is the current behaviour ;)